mountinfo/mountinfo_filters_test.go

```diff
@@ -0,0 +1,36 @@
+package mountinfo
+
+import "testing"
+
+func TestPrefixFilter(t *testing.T) {
+	tests := []struct {
+		prefix     string
+		mountPoint string
+		shouldSkip bool
+	}{
+		{prefix: "/a", mountPoint: "/a", shouldSkip: false},
+		{prefix: "/a", mountPoint: "/a/", shouldSkip: false},
+		{prefix: "/a", mountPoint: "/a/b", shouldSkip: false},
+		{prefix: "/a", mountPoint: "/a/b/", shouldSkip: false},
+		{prefix: "/a", mountPoint: "/aa", shouldSkip: true},
+		{prefix: "/a", mountPoint: "/aa/", shouldSkip: true},
+		{prefix: "/a", mountPoint: "/aa/b", shouldSkip: true},
+		{prefix: "/a", mountPoint: "/aa/b/", shouldSkip: true},
+
+		// invalid prefix: prefix path must be cleaned and have no trailing slash
+		{prefix: "/a/", mountPoint: "/a", shouldSkip: true},
+		{prefix: "/a/", mountPoint: "/a/b", shouldSkip: true},
+		{prefix: "/a/", mountPoint: "/a/b/", shouldSkip: true},
```

Comment:
@kolyshkin I made the change you suggested in https://github.com/moby/sys/pull/61#discussion_r577260268, but I kept the `m.Mountpoint != prefix`. I wasn't sure if mountinfo could return paths with a trailing slash (at least we don't do a `path.Clean()` in our code.

If we're sure mountinfo will _never_ return a trailing slash, then we can remove some of these tests;

```suggestion
		{prefix: "/a", mountPoint: "/a", shouldSkip: false},
		{prefix: "/a", mountPoint: "/a/b", shouldSkip: false},
		{prefix: "/a", mountPoint: "/aa", shouldSkip: true},
		{prefix: "/a", mountPoint: "/aa/b", shouldSkip: true},

		// invalid prefix: prefix path must be cleaned and have no trailing slash
		{prefix: "/a/", mountPoint: "/a", shouldSkip: true},
		{prefix: "/a/", mountPoint: "/a/b", shouldSkip: true},
```

Reply:
I am pretty sure mountinfo never returns something with an ending slash; IOW we can assume that the paths it provides are as clean as `filepath.Clean()` makes them.

Reply:
Thx; I'll remove those case